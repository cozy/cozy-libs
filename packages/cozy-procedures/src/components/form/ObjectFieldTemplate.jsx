import React from 'react'
import PropTypes from 'prop-types'

const ObjectFieldTemplate = props => (
  <div>
    {props.properties.map(element => {
      return (
        <div className="property-wrapper" key={element.name}>
          {element.content}
        </div>
      )
    })}
  </div>
)

ObjectFieldTemplate.propTypes = {
  properties: PropTypes.arrayOf(
    PropTypes.shape({
      content: PropTypes.node.isRequired,
      name: PropTypes.string.isRequired
    })
  ).isRequired
}

export default ObjectFieldTemplate
