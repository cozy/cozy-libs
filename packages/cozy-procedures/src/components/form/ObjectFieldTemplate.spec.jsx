import React from 'react'

import ObjectFieldTemplate from './ObjectFieldTemplate'

describe('ObjectFieldTemplate component', () => {
  it('should match snapshot', () => {
    const props = {
      properties: [
        {
          content: <span>Foo</span>,
          name: 'foo'
        },
        {
          content: <span>Bar</span>,
          name: 'bar'
        }
      ]
    }
    const component = shallow(
      <ObjectFieldTemplate {...props}>
        <input name="foobar" />
      </ObjectFieldTemplate>
    )
    expect(component).toMatchSnapshot()
  })
})
