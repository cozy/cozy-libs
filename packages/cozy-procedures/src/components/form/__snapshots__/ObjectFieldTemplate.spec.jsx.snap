// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ObjectFieldTemplate component should match snapshot 1`] = `
<div>
  <div
    className="property-wrapper"
    key="foo"
  >
    <span>
      Foo
    </span>
  </div>
  <div
    className="property-wrapper"
    key="bar"
  >
    <span>
      Bar
    </span>
  </div>
</div>
`;
